torch/_higher_order_ops/cond.py

```diff
@@ -28,11 +30,139 @@
+                    elif isinstance(t, torch.SymBool):
+                        return t.node._hint
+                    elif isinstance(t, torch.fx.proxy.Proxy):
+                        raise RuntimeError(
+                            f"Unable to symbolically trace HigherOrderOperators {op}"
+                        )
+                    return t
+
+                args = (args[0], args[1], args[2], tuple(args[3]))
+                with fake_tensor_mode:
+                    new_args = pytree.tree_map(from_fun, args)
+
+                    # we need to wrap true_fn/false_fn up otherwise the local scope
+                    # will contain the original true_fn and false_fn's signature
+                    def wrapper(new_args):
+                        return cond(*new_args)
+
+                    # need to do it together otherwise will need to merge input ->
+                    # duplicated effort with what we have already
+                    with _turn_off_is_fx_tracing():
+                        expo_result = torch._dynamo.export(
+                            wrapper, rewrite_sig=False, fake_mode=fake_tensor_mode
+                        )(new_args)
+                    gm, guards = expo_result
+                    example_inputs = expo_result.example_inputs
+                    example_inputs_ids = [id(inp) for inp in example_inputs]
+                    id_to_name = {
+                        id(guard.obj_weakref()): guard.name
+                        for guard in guards
+                        if guard.obj_weakref is not None
+                        and id(guard.obj_weakref()) in example_inputs_ids
+                    }
+                    example_names = [id_to_name[id(inp)] for inp in example_inputs]
+
+    # fake the new_args with original args
+    local_scope = {"L": {**locals(), "new_args": args}, "G": globals()}
+    pos_args = [eval(name, {}, local_scope) for name in example_names]
+
+    # We need to extract true_gm and false_gm from export
+    # as export won't add sym bool
+    def bind_branch_and_args(gm, pos_args):
+        ph2orig = dict(
+            zip((ph for ph in gm.graph.nodes if ph.op == "placeholder"), pos_args)
+        )
+        cond_node = next((n for n in gm.graph.nodes if n.target is cond), None)
+        assert cond_node
+        true_gm = getattr(gm, cond_node.args[1].name)
+        false_gm = getattr(gm, cond_node.args[2].name)
+        pos_args = []
+        for arg_node in cond_node.args[3]:
+            if arg_node.op == "placeholder" and arg_node in ph2orig:
+                pos_args.append(ph2orig[arg_node])
+            elif arg_node.op == "get_attr":
+                pos_args.append(getattr(gm, arg_node.target))
+            else:
+                raise RuntimeError(f"Cannot bind to original argumentes for {arg_node}")
+        return true_gm, false_gm, tuple(pos_args)
+
+    return cond(args[0], *bind_branch_and_args(gm, pos_args))
```

Comment:
TODO(rzou): read through getattr

Reply:
Open question if this is good or if we should use the "trace through NN modules config option"